components/Onboarding/Page.js

```diff
@@ -0,0 +1,166 @@
+import { withRouter } from 'next/router'
+import { css } from 'glamor'
+
+import {
+  mediaQueries,
+  fontStyles,
+  Center,
+  Interaction,
+  linkRule,
+  Button,
+  colors
+} from '@project-r/styleguide'
+
+import Greeting from './Greeting'
+import Newsletter from './Sections/Newsletter'
+import AppLogin from './Sections/AppLogin'
+import Usability from './Sections/Usability'
+import Profile from './Sections/Profile'
+import Frame from '../Frame'
+import { scrollIt } from '../../lib/utils/scroll'
+import { HEADER_HEIGHT } from '../constants'
+import { Link } from '../../lib/routes'
+
+const { P } = Interaction
+
+const HEADING_FONT_SIZE = 22
+
+const styles = {
+  title: css({
+    ...fontStyles.sansSerifMedium58,
+    textAlign: 'center',
+    marginTop: 40,
+    marginBottom: 80,
+    [mediaQueries.onlyS]: {
+      ...fontStyles.sansSerifMedium40,
+      marginBottom: 50
+    }
+  }),
+  p: css({
+    marginBottom: 20
+  }),
+  sections: css({
+    paddingTop: HEADING_FONT_SIZE,
+    paddingBottom: HEADING_FONT_SIZE
+  }),
+  buttonContainer: css({
+    marginBottom: HEADING_FONT_SIZE
+  })
+}
+
+class Page extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      expandedSection: null,
+      hasOnceVisitedAll: false
+    }
+
+    this.sections = [
+      { component: Newsletter, name: 'newsletter', ref: React.createRef(), visited: false },
+      { component: AppLogin, name: 'app-login', ref: React.createRef(), visited: false },
+      { component: Usability, name: 'usability', ref: React.createRef(), visited: false },
+      { component: Profile, name: 'profile', ref: React.createRef(), visited: false }
+    ]
+
+    this.onExpand = (props) => {
+      this.setState({ expandedSection: props.name })
+    }
+
+    this.onContinue = () => {
+      const { expandedSection } = this.state
+      let sectionIndex = 0
+
+      if (expandedSection) {
+        const currentSection = this.sections.find(({ name }) => expandedSection === name)
+        currentSection.visited = true
+
+        const nextIndex = this.sections.findIndex(({ name }) => expandedSection === name) + 1
+
+        if (nextIndex < this.sections.length) {
+          sectionIndex = nextIndex
+        } else {
+          this.setState({
+            expandedSection: null,
+            hasOnceVisitedAll: this.sections.every(section => !!section.visited)
+          })
+
+          return
+        }
+      }
+
+      this.setState(
+        { expandedSection: this.sections[sectionIndex].name },
+        () => {
+          const { top } = this.sections[sectionIndex].ref.current.getBoundingClientRect()
+          const { pageYOffset } = window
+
+          const target = pageYOffset + top - (HEADER_HEIGHT * 1.2)
+          scrollIt(target, 400)
+        }
+      )
+    }
+  }
+
+  render () {
+    const meta = {
+      title: 'Willkommen bei der Republik'
+    }
+
+    const origin = this.props.router.query.origin || 'default'
```

Comment:
- [ ] Limit allowed `origin` inputs to "claim", "access", "package", "other", "default"